### ISSUE TITLE
Batch inference for ITM

### ISSUE THREAD

**Issue by fzhjeffrey** [salesforce/LAVIS#424]
I have multiple images and one text and I want to measure the ITM score and sort the images based on their similarity to the single text, Is there a batch inference option in BLIP2? Thanks!

**Comment by dxli94** [comment]
The model should support batch inference by itself. Just duplicate the text as a list. Then feed the model with batched images and text list. This will give you a list of matching scores as output.

**Comment by fzhjeffrey** [comment]
It works! Thank you so much!

**Comment by Keep-lucky** [comment]
> The model supports batch inference by itself. Just duplicate the text as a list. Then feed the model with batched images and text list. This will give you a list of matching scores as output.该模型本身支持批量推理。只需将文本复制为列表即可。然后向模型提供批量图像和文本列表。这将为您提供匹配分数的列表作为输出。
Sorry to bother you，I also encountered the same problem, would you please share the code? Thank you!

**Comment by bexxnaz** [comment]
Hi, and I changed the padding to True in line 64 (lavis/models/blip2_models/blip2_image_text_matching.py).  The updated code is as follows:
```
text = self.tokenizer(
    caption,
    truncation=True,
    max_length=self.max_txt_len,
    padding=True,
    return_tensors="pt",
).to(image.device)

```

Batch inference for ITM:

```
model, vis_processors, text_processors = load_model_and_preprocess("blip2_image_text_matching", "coco", device=device, is_eval=True)

class ImageCaptionDataset(Dataset):
    def __init__(self, image_root, ann_root):
        self.annotation = json.load(open(ann_root,'r'))
        self.img_dir = image_root
        self.img_names = os.listdir(self.img_dir)
        self.vis_processors = vis_processors
        self.text_processors = text_processors

    def __len__(self):
        return len(self.annotation)

    def __getitem__(self, index):
        ann = self.annotation[index]
        image_path = os.path.join(self.img_dir,ann['image_id'])
        image = Image.open(image_path).convert('RGB')
        caption = ann['blip2']
        img = self.vis_processors["eval"](image).unsqueeze(0)
        txt = self.text_processors["eval"](caption)

        data = {'img':img, 'txt':txt }
        return data

def my_collate_fn(batch):
    images = []
    captions = []
    images = torch.stack([item['img'] for item in batch])
    captions = [item['txt'] for item in batch]
    return {'text_input': captions, 'image':images}

dataset = ImageCaptionDataset(img_dir,ann_root)
dataloader = DataLoader(dataset, batch_size=8, num_workers=1, collate_fn=my_collate_fn)

for batch in dataloader:
    batch['image'] = batch['image'].squeeze(1).to(device)
    itc_output = model({"image": batch['image'], "text_input":  batch['text_input']}, match_head="itc")
    itm_output = model({"image": batch['image'], "text_input":  batch['text_input']}, match_head="itm")

```
